<?xml version="1.0"?>
<robot name="gripper">

  <!-- Gripper base link with STL mesh -->
  <link name="gripper_base">
    <!-- Gripper mesh scaled from mm to meters -->
    <visual>
      <origin xyz="0 0 0" rpy="3.14159 0 0"/>  <!-- Flipped 180 degrees around X axis -->
      <geometry>
        <mesh filename="package://motion_control_module/meshes/gripper.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="gripper_material">
        <color rgba="0.7 0.7 0.7 1.0"/>  <!-- Light gray -->
      </material>
    </visual>

    <collision>
      <origin xyz="0 0 0" rpy="3.14159 0 0"/>  <!-- Flipped 180 degrees around X axis -->
      <geometry>
        <mesh filename="package://motion_control_module/meshes/gripper.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>

    <inertial>
      <origin xyz="0 0 0.05" rpy="0 0 0"/>
      <mass value="0.5"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0"
               iyy="0.001" iyz="0.0"
               izz="0.001"/>
    </inertial>
  </link>

  <!-- Gripper tip frame for easy reference -->
  <link name="gripper_tip"/>

  <joint name="gripper_tip_joint" type="fixed">
    <parent link="gripper_base"/>
    <child link="gripper_tip"/>
    <origin xyz="0 0 0.15" rpy="0 0 0"/>  <!-- Adjust Z value based on gripper length -->
  </joint>

</robot>
